packages/react-native-audio-api/common/cpp/audioapi/core/effects/ConvolverNode.cpp

```diff
@@ -0,0 +1,253 @@
+    }
+    power += channelPower;
+  }
+
+  power = std::sqrt(power / (numberOfChannels * length));
+  if (power < MIN_IR_POWER) {
+    power = MIN_IR_POWER;
+  }
+  scaleFactor_ = 1 / power;
+  scaleFactor_ *= std::pow(10, GAIN_CALIBRATION * 0.05f);
+  scaleFactor_ *= gainCalibrationSampleRate_ / buffer_->getSampleRate();
+}
+
+void ConvolverNode::performConvolution(
+    const std::shared_ptr<AudioBus> &processingBus) {
+  std::vector<std::thread> threads;
+  if (processingBus->getNumberOfChannels() == 1) {
+    if (convolvers_.size() == 1) {
+      convolvers_[0].process(
+          processingBus->getChannel(0)->getData(),
+          intermediateBus_->getChannel(0)->getData());
+    } else if (convolvers_.size() == 2) {
+      for (int i = 0; i < convolvers_.size(); ++i) {
+        threads.emplace_back([this, i, processingBus]() {
+          convolvers_[i].process(
+              processingBus->getChannel(0)->getData(),
+              intermediateBus_->getChannel(i)->getData());
+        });
+      }
+    } else { // 4 channel IR
+      for (int i = 0; i < 2; ++i) {
+        threads.emplace_back([this, i, processingBus]() {
+          convolvers_[i].process(
+              processingBus->getChannel(0)->getData(),
+              intermediateBus_->getChannel(i)->getData());
+        });
+      }
+      threads.emplace_back([this, processingBus]() {
+        convolvers_[2].process(
+            processingBus->getChannel(0)->getData(),
+            intermediateBus_->getChannel(2)->getData());
+      });
+      threads.emplace_back([this, processingBus]() {
+        convolvers_[3].process(
+            processingBus->getChannel(0)->getData(),
+            intermediateBus_->getChannel(3)->getData());
+      });
+    }
+  } else if (processingBus->getNumberOfChannels() == 2) {
+    if (convolvers_.size() == 2) {
+      for (int i = 0; i < 2; ++i) {
+        threads.emplace_back([this, i, processingBus]() {
+          convolvers_[i].process(
+              processingBus->getChannel(i)->getData(),
+              intermediateBus_->getChannel(i)->getData());
+        });
+      }
+    } else { // 4 channel IR
+      threads.emplace_back([this, processingBus]() {
+        convolvers_[0].process(
+            processingBus->getChannel(0)->getData(),
+            intermediateBus_->getChannel(0)->getData());
+      });
+      threads.emplace_back([this, processingBus]() {
+        convolvers_[1].process(
+            processingBus->getChannel(0)->getData(),
+            intermediateBus_->getChannel(3)->getData());
+      });
+      threads.emplace_back([this, processingBus]() {
+        convolvers_[2].process(
+            processingBus->getChannel(1)->getData(),
+            intermediateBus_->getChannel(2)->getData());
+      });
+      threads.emplace_back([this, processingBus]() {
+        convolvers_[3].process(
+            processingBus->getChannel(1)->getData(),
+            intermediateBus_->getChannel(1)->getData());
+      });
+    }
+  }
```

Comment:
```suggestion
  } else if (processingBus->getNumberOfChannels() == 2) {
   std::vector<int> inputChannelMap;
   std::vector<int> outputChannelMap;

   if (convolvers_.size() == 2) {
       inputChannelMap  = {0, 1};
       outputChannelMap = {0, 1};
   } else { // 4 channel IR
       inputChannelMap  = {0, 0, 1, 1};
       outputChannelMap = {0, 3, 2, 1};
   }

   for (size_t i = 0; i < convolvers_.size(); ++i) {
         threads.emplace_back([this, i, &processingBus, &inputChannelMap, &outputChannelMap]() {
              convolvers_[i].process(
                  processingBus->getChannel(inputChannelMap[i])->getData(),
                  intermediateBus_->getChannel(outputChannelMap[i])->getData());
        });
     }
  }
```
 I think this version is a bit more readable